cuda/matrix/ell_kernels.cu

```diff
@@ -111,12 +140,21 @@ __global__ __launch_bounds__(default_block_size) void abstract_spmv(
 {
     const ValueType alpha_val = alpha[0];
     const ValueType beta_val = beta[0];
-    spmv_kernel(
-        num_rows, val, col, stride, num_stored_elements_per_row, b, b_stride, c,
-        c_stride,
-        [&alpha_val, &beta_val](const ValueType &x, const ValueType &y) {
-            return alpha_val * x + beta_val * y;
-        });
+    if (atomic) {
+        spmv_kernel<subwarp_size, atomic>(
+            num_rows, val, col, stride, num_stored_elements_per_row, b,
+            b_stride, c, c_stride,
+            [&alpha_val](const ValueType &x, const ValueType &y) {
+                return alpha_val * x;
+            });
+    } else {
+        spmv_kernel<subwarp_size, atomic>(
+            num_rows, val, col, stride, num_stored_elements_per_row, b,
+            b_stride, c, c_stride,
+            [&alpha_val, &beta_val](const ValueType &x, const ValueType &y) {
+                return alpha_val * x + beta_val * y;
```

Comment:
Can you comment on these functions somewhere, why are they different in all of those cases, what do they do?